extensions/control-plane/provision/provision-gcs/src/test/java/org/eclipse/edc/connector/provision/gcp/GcsConsumerResourceDefinitionGeneratorTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2022 ZF Friedrichshafen AG
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ *    ZF Friedrichshafen AG - Initial API and Implementation
+ */
+
+package org.eclipse.edc.connector.provision.gcp;
+
+import org.eclipse.edc.connector.transfer.spi.types.DataRequest;
+import org.eclipse.edc.gcp.storage.GcsStoreSchema;
+import org.eclipse.edc.policy.model.Policy;
+import org.eclipse.edc.spi.types.domain.DataAddress;
+import org.eclipse.edc.spi.types.domain.asset.Asset;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class GcsConsumerResourceDefinitionGeneratorTest {
+
+    private final Pattern regexPattern = Pattern.compile("([a-f0-9]{8}(-[a-f0-9]{4}){4}[a-f0-9]{8})");
+    private GcsConsumerResourceDefinitionGenerator generator;
+
+    @BeforeEach
+    void setUp() {
+        generator = new GcsConsumerResourceDefinitionGenerator();
+    }
+
+    @Test
+    void generate() {
```

Comment:
actually they can't be null, so I won't bother testing these cases

Reply:
At the very least I would expect an `Object.requireNotNull` to assert that, and a test for it. testing only the happy path is dangerous.

Reply:
We don't do that in any other component I think, as I wrote in the last comment, it would be useless defensive stuff in a component that would never ever be called with null arguments.

Reply:
The Generators are already covered with the `Object.requireNotNull` so that no method can receive null parameters